ClavaWeaver/resources/clava/test/weaver/PragmaData.lara

```diff
@@ -30,19 +30,25 @@ aspectdef PragmaData
 
 	// Setting data without pragma creates a pragma
 	var $loopWithoutPragma = Query.search("function", "insertWithoutPragma").first();
-	$loopWithoutPragma.data.a = 42;
-	$loopWithoutPragma.data.b = 43;	
+	var obj = $loopWithoutPragma.data;
+	obj.a = 42;
+	obj.b = 43;
+	$loopWithoutPragma.setData(obj);
 	println("Loop without pragma after setting data: " + $loopWithoutPragma.pragmas.map(p => p.code));
 
 	// Setting already existing data updates pragma
 	var $loopWithUpdatedPragma = Query.search("function", "updatePragma").first();
-	$loopWithUpdatedPragma.data.a = 200;
+	obj = $loopWithUpdatedPragma.data;
+	obj.a = 200;
+	$loopWithUpdatedPragma.setData(obj);
 	println("Loop with updated pragma: " + $loopWithUpdatedPragma.pragmas.map(p => p.code));
 
 	// Information persists after rebuilds
 	var $loopBeforeRebuild = Query.search("function", "rebuild").first();
 	println("Is parallel before set and rebuild: " + $loopBeforeRebuild.data.isParallel);
-	$loopBeforeRebuild.data.isParallel = true;
+	obj = $loopBeforeRebuild.data;
+	obj.isParallel = true;
+	$loopBeforeRebuild.setData(obj);
 
```

Comment:
This is a breaking change, add an issue to solve this by using a proxy object that communicates the changes to data in Java.